### PR TITLE
Add table modal for report link widget

### PR DESCRIPTION
**Pull request by shubhamsc**

### CODE REVIEW

File: src/tables/table-loader.component.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  it(`should render Loading message when table is loading`, done => {
+    mockEsmAPI.openmrsFetch.mockResolvedValueOnce(mockTableData);
+    const { queryByText } = render(
+      <TableLoader {...commonWidgetProps} config={mockTableData} />
+    );
+    expect(queryByText("Loading...")).not.toBeNull();
+    done();
+  });
+
```
Comment (rrameshbtech) [inline src/tables/table-loader.component.test.tsx]: Can we have a test for checking positive flow where table is rendering?

File: src/report-links/report-links.css [hunk 2]
```diff
@@ -38,4 +38,6 @@
   font-size: 16px;
   color: #e01e5c;
   padding: 0px;
+  right: 30px;
+  position: fixed;
```
[the remark this answers is not in the document]
Comment (shubhamsc) [reply src/report-links/report-links.css]: does not need to set position explicit.

File: src/tables/table-loader.component.test.tsx (hunk 1 above)
Comment (shubhamsc) [reply src/tables/table-loader.component.test.tsx]: added test for checking react table

File: src/tables/table-loader.component.test.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  it(`should render react table when table is loading`, done => {
```
Comment (rrameshbtech) [inline src/tables/table-loader.component.test.tsx]: Can this be better named like `should render response as table with valid report response`

File: src/tables/table-loader.component.test.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    mockEsmAPI.openmrsFetch.mockResolvedValueOnce(mockTableData);
+    const { container } = render(
+      <TableLoader {...commonWidgetProps} config={mockTableData} />
+    );
+
+    expect(container.getElementsByClassName("ReactModalPortal")).toBeTruthy();
```
Comment (rrameshbtech) [inline src/tables/table-loader.component.test.tsx]: Assert the content also with checking whether the table is shown. This test will pass even if the table is showing `no record found`. But that is invalid.

File: src/tables/table-loader.component.test.tsx (hunk 3 above)
Comment (shubhamsc) [reply src/tables/table-loader.component.test.tsx]: renamed test

File: src/tables/table-loader.component.test.tsx (hunk 4 above)
Comment (shubhamsc) [reply src/tables/table-loader.component.test.tsx]: added test for showing table data

**Comment by mddubey** [comment]
LGTM.

**Pull request merged**